packages/tpetra/core/src/Tpetra_Details_FixedHashTable_Cuda.cpp

```diff
@@ -79,6 +79,14 @@ namespace Details {
 
   TPETRA_INSTANTIATE_L( TPETRA_DETAILS_FIXEDHASHTABLE_INSTANT_CUDA_INT )
 
+  // Make sure that KeyType = long, ValueType = int and
+  // KeyType = long long, ValueType = int both get instantiated -
+  // whenever Tpetra is enabled, Xpetra::Epetra_Map depends on both
+  TPETRA_DETAILS_FIXEDHASHTABLE_INSTANT(long, int, cuda_device_type)
```

Comment:
1. You actually want `int, long`, not the other way around.  The macro (see the bottom of `Tpetra_Details_FixedHashTable_def.hpp`) takes arguments `LO, GO, DEVICE`.  It reverses `LO` and `GO` as template arguments to `FixedHashTable`.

2. These macro invocations don't belong inside `#ifndef HAVE_TPETRA_INST_INT_INT ... #endif`.   You should actually put them inside `#ifndef HAVE_TPETRA_INST_INT_LONG ... #endif` resp. `#ifndef HAVE_TPETRA_INST_INT_LONG_LONG ... #endif`.  Otherwise, you'll get link errors due to duplicate instantiations.  (Are you somehow testing with ETI OFF?)
 
3. I know it's not your doing, but it seems really broken that Xpetra needs the `long` instantiation for Epetra.

4. Why are we instantiating both for `cuda_uvm_device_type` and for `cuda_device_type`?  Do we really need both?